### DISCUSSION TITLE
Rotate

### DISCUSSION THREAD

**Discussion by Fuffret** [lfroms/viewfinder#16] [Ideas]
I think it would be super useful if you could add a way to rotate the image coming from the camera. This could be used for people who mount their webcam on the bottom oof their screen or for those who rotate their monitor often.

**Comment by lfroms** [comment]
Great suggestion!  I'll look into the feasibility of this, since support for this kind of rotation largely depends on the camera used and what APIs it supports.  We would need to control the camera itself (or perhaps use AVFoundation's [`videoOrientation`](https://developer.apple.com/documentation/avfoundation/avcaptureconnection/1389415-videoorientation)) to put it into this state since the video signal itself isn't passing directly through the Viewfinder app for processing (it's just a controller).

To confirm, is the goal to be able to flip it to be upside-down?

**Reply by Fuffret** [reply]
Ideally rotate like 90 or 270 degrees, i attempted to  represent what I mean  below (hopefully it turns out as intended)
0                               90              180                           270
—————————    |            .   |  —————————   |   .           |
   `     `     `    `         |            .   |                                  |  .            |
                                 |            .   |      ,     ,     ,     ,         |  .            |
—————————    |            .   | —————————    |   .           |

**Reply by Fuffret** [reply]
It indeed came out like a dumpster fire here is a picture attached
![D8709392-B756-4368-922C-16DDE05492B4](https://user-images.githubusercontent.com/105463581/168195100-cd3a7444-fc74-46fd-8f74-5278d8307010.jpeg)

**Reply by lfroms** [reply]
Thanks for the illustration!  I will look into this as part of future work. 👍

**Reply by Fuffret** [reply]
Thanks
